portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,31 +15,61 @@
 package com.google.sps.servlets;
 
 import com.google.gson.Gson;
-import com.google.sps.data.Comments;
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns comments */
+
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private final Comments comments = new Comments();
   private final Gson GSON_OBJECT = new Gson();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    final String userCommentLimit = request.getParameter("comment-limit");
+    final int commentLimit = Integer.parseInt(userCommentLimit); 
+    final List<Comment> comments = new ArrayList<>();
+    for (final Entity entity : results.asIterable()) {
+      if (comments.size() == commentLimit) {
```

Comment:
Is there a Datastore query param that allows you to specify the limit, so that you don't have to query everything and do a for-loop to limit the size here?

Reply:
yes, found a tool to do this (imported fetchoptions)